.github/workflows/go.yml

```diff
@@ -2,61 +2,69 @@ name: Go
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
   pull_request:
-    branches: [ master ]
+    branches: [master]
 
 jobs:
   main:
     runs-on: ubuntu-18.04
     strategy:
       matrix:
-        go: [ '1.15' ]
+        go: ["1.15"]
     name: Go ${{ matrix.go }}
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v2
-      id: go
-      with:
-        go-version: ${{ matrix.go }}
-
-    - name: Verify modules
-      run: go mod verify
-
-    - name: Verify format
-      run: |
-        make fmt
-        git diff --exit-code
-
-    - name: Verify manifests
-      run: |
-        make manifests
-        git diff --exit-code
-
-    - name: Verify imported manifests
-      run: |
-        make import-manifests
-        git diff --exit-code
-
-    - name: Verify bindata
-      run: |
-        make verify-bindata
-
-    - name: Create K8s KinD Cluster
-      run: make test-cluster
-
-    - name: Build and Push Test Container Image to Local Insecure Container Registry
-      run: |
-        make docker-build IMG=localhost:5000/gatekeeper-operator:$GITHUB_SHA
-        make docker-push IMG=localhost:5000/gatekeeper-operator:$GITHUB_SHA
-
-    - name: Unit and Integration Tests
-      run: make test
-
-    - name: Deploy Gatekeeper Operator
-      run: |
-        make install
-        make deploy NAMESPACE=mygatekeeper IMG=localhost:5000/gatekeeper-operator:$GITHUB_SHA
-
-    - name: E2E Tests
-      run: make test-e2e
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        id: go
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Verify modules
+        run: go mod verify
+
+      - name: Verify format
+        run: |
+          make fmt
+          git diff --exit-code
+
+      - name: Verify manifests
+        run: |
+          make manifests
+          git diff --exit-code
+
+      - name: Verify imported manifests
+        run: |
+          make import-manifests
+          git diff --exit-code
+
+      - name: Verify bindata
+        run: |
+          make verify-bindata
+
+      - name: Create K8s KinD Cluster
+        run: kind create cluster
+
+      - name: Build and Push Test Container Image to KIND node
+        run: |
+          make docker-build IMG=localhost:5000/gatekeeper-operator:$GITHUB_SHA
+          kind load docker-image localhost:5000/gatekeeper-operator:$GITHUB_SHA
+
+      - name: Unit and Integration Tests
+        run: make test
+
+      - name: Deploy Gatekeeper Operator
+        run: |
+          make install
+          kubectl create namespace gatekeeper-system 
+          kubectl create namespace mygatekeeper
+          make deploy NAMESPACE=gatekeeper-system IMG=localhost:5000/gatekeeper-operator:$GITHUB_SHA
+          kubectl patch deployment -n gatekeeper-system gatekeeper-operator-controller-manager -p '{"spec": {"template": {"spec":{"containers":[{"name":"manager","imagePullPolicy":"IfNotPresent"}]}}}}'
```

Comment:
@font Since we are storing the image on the KIND node, we need to tell Kubernetes to not attempt to retrieve the image from a remote source and use the image as it is on the node